From 811948171bff15bb47278aba25edcb5f34de12d2 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 24 Apr 2018 13:13:05 +0800
Subject: [PATCH 3690/5242] MLK-18123-4 gpu: imx: imx8_dprc: Determine if
 NV12/21 is supported for DC0 BLIT1

commit  348bb0cef3cc711ece61a9935b7bf09499f59e42 from
https://source.codeaurora.org/external/imx/linux-imx.git

With the DPR vs PRG muxing introduced in silicon, the DPR channel of
DC0 BLIT1 shares auxiliary PRG with the DPR channel of DC0 BLIT0.
So, the DPR channel of DC0 BLIT1 may have chance to fetch two planar
NV12/21 frames by using two PRGs(one primary and one auxiliary).
Let's reflect this in the dprc_format_supported() helper.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 562f253..431649f 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -714,6 +714,12 @@ bool dprc_format_supported(struct dprc *dprc, u32 format, u64 modifier)
 		case SC_R_DC_0_WARP:
 		case SC_R_DC_1_WARP:
 			return false;
+		case SC_R_DC_0_BLIT1:
+			if (!dprc->devtype->has_fixup)
+				break;
+
+			return (modifier == DRM_FORMAT_MOD_NONE ||
+				modifier == DRM_FORMAT_MOD_AMPHION_TILED);
 		}
 		return (dprc->has_aux_prg &&
 			(modifier == DRM_FORMAT_MOD_NONE ||
-- 
1.7.9.5

